FormalConjectures/Paper/StrongSensitivityConjecture.lean

```diff
@@ -0,0 +1,132 @@
+  by Hao Huang (see Section 3, Concluding Remarks)
+* [Variations on the Sensitivity Conjecture](https://arxiv.org/abs/1011.0354)
+  by Pooya Hatami, Raghav Kulkarni, and Denis Pankratov (see Question 3.1)
+* [On the Degree of Boolean Functions as Real Polynomials](https://link.springer.com/article/10.1007/BF01263419)
+  by Noam Nisan, and Mario Szegedy (see Section 4, Open Problems)
+-/
+
+namespace StrongSensitivityConjecture
+
+open Finset Function Classical
+
+section Sensitivity
+
+variable {n : ℕ}
+
+/-- Flip operator,
+`flip x B` returns input `x` with bits in block `B` inverted. -/
+def flip (x : Fin n → Bool) (B : Finset (Fin n)) : Fin n → Bool :=
+  fun i => if i ∈ B then !(x i) else x i
+
+/-- Local sensitivity s(f,x),
+number of indices where flipping one bit changes the value of `f`. -/
+def sensitivityAt (f : (Fin n → Bool) → Bool) (x : Fin n → Bool) : ℕ :=
+  #{i | f (flip x {i}) ≠ f x}
+
+/-- Global sensitivity s(f),
+maximum sensitivity of `f` over all inputs. -/
+def sensitivity (f : (Fin n → Bool) → Bool) : ℕ :=
+  univ.sup (sensitivityAt f)
+
+/-- Check validity of block collection (disjoint and sensitive),
+A collection of blocks `cB` is valid for `f` at `x` if the blocks are
+disjoint and flipping any block changes `f(x)`. -/
+def IsValidBlockConfig (f : (Fin n → Bool) → Bool) (x : Fin n → Bool)
+    (cB : Finset (Finset (Fin n))) : Prop :=
+  (cB : Set (Finset (Fin n))).PairwiseDisjoint id ∧
+  ∀ B ∈ cB, f (flip x B) ≠ f x
+
+/-- Local block sensitivity bs(f,x),
+maximum size of a collection of sensitive, disjoint blocks for `f` at `x`. -/
+noncomputable def blockSensitivityAt (f : (Fin n → Bool) → Bool) (x : Fin n → Bool) : ℕ :=
+  Finset.sup {cB | IsValidBlockConfig f x cB} card
+
+/-- Global block sensitivity of `f`,
+maximum block sensitivity of `f` over all inputs. -/
+noncomputable def blockSensitivity (f : (Fin n → Bool) → Bool) : ℕ :=
+  univ.sup (blockSensitivityAt f)
+
+/-- Strong Sensitivity Conjecture,
+for every Boolean function `f : {0,1}^n → {0,1}`,
+`bs(f) ≤ s(f)^2`.
+
+We call this the *strong* sensitivity conjecture because the original sensitivity
+conjecture only asked for a polynomial bound in terms of `s(f)`. Huang's
+celebrated result (often called the sensitivity theorem) gives a quartic bound,
+`bs(f) ≤ s(f)^4`, thereby settling the original conjecture. -/
+@[category research open, AMS 68]
+theorem strong_sensitivity_conjecture {n : ℕ} (f : (Fin n → Bool) → Bool) :
+    blockSensitivity f ≤ sensitivity f ^ 2 := by
+  sorry
+
+
+/-- Simple test example,
+A Boolean function whose block sensitivity is strictly greater than
+its sensitivity. Source: [Nisan1989](https://dl.acm.org/doi/10.1145/73007.73038).
+
+`nisanExample(x) = 1` iff the Hamming weight of `x` is either
+`n/2` or `n/2 + 1`. We assume `n` is a multiple of 4.
+The function is symmetric, so its value only depends on the Hamming weight
+of the input. -/
+@[category test]
```

Comment:
Generally speaking we reserve the `test` category to theorems (though this particular use case is understandable)